src/filters/oidc/jwks_resolver.h

```diff
@@ -109,6 +113,43 @@ class DynamicJwksResolverImpl : public JwksResolver {
   absl::Mutex mux_;
 };
 
+class JwksResolverCache {
+ public:
+  JwksResolverCache(const authservice::config::oidc::OIDCConfig& config,
+                    boost::asio::io_context& ioc)
+      : config_(config) {
+    switch (config_.jwks_config_case()) {
+      case config::oidc::OIDCConfig::kJwks:
+        resolver_ =
+            std::make_shared<oidc::StaticJwksResolverImpl>(config_.jwks());
+        break;
+      case config::oidc::OIDCConfig::kJwksFetcher: {
+        uint32_t periodic_fetch_interval_sec =
+            config_.jwks_fetcher().periodic_fetch_interval_sec();
+        if (periodic_fetch_interval_sec == 0) {
+          periodic_fetch_interval_sec = 1200;
```

Comment:
make this a constant var like `kJwksPeriodicFetchIntervalSec`, versus in-place numeric value?